packages/tldraw/src/lib/shapes/shared/useEditableText.ts

```diff
@@ -44,75 +37,18 @@ export function useEditableText(id: TLShapeId, type: string, text: string) {
 	useEffect(() => {
 		if (!isEditing) return
 
-		const elm = rInput.current
-		if (!elm) return
-
-		// Focus if we're not already focused
-		if (document.activeElement !== elm) {
-			elm.focus()
-
-			// On mobile etc, just select all the text when we start focusing
-			if (editor.getInstanceState().isCoarsePointer) {
-				elm.select()
-			}
-		} else {
-			// This fixes iOS not showing the cursor sometimes. This "shakes" the cursor
-			// awake.
-			if (editor.environment.isSafari) {
-				elm.blur()
-				elm.focus()
-			}
+		if (editor.getInstanceState().isCoarsePointer) {
+			rInput.current?.select()
 		}
 
-		// When the selection changes, save the selection ranges
-		function updateSelection() {
-			const selection = window.getSelection?.()
-			if (selection && selection.type !== 'None') {
-				const ranges: Range[] = []
-				for (let i = 0; i < selection.rangeCount; i++) {
-					ranges.push(selection.getRangeAt?.(i))
-				}
-				rSelectionRanges.current = ranges
-			}
-		}
-
-		document.addEventListener('selectionchange', updateSelection)
-		return () => {
-			document.removeEventListener('selectionchange', updateSelection)
+		// XXX(mime): This fixes iOS not showing the cursor sometimes.
+		// This "shakes" the cursor awake.
+		if (editor.environment.isSafari) {
+			rInput.current?.blur()
+			rInput.current?.focus()
 		}
 	}, [editor, isEditing])
 
-	// 2. Restore the selection changes (and focus) if the element blurs
-	// When the label blurs, deselect all of the text and complete.
-	// This makes it so that the canvas does not have to be focused
-	// in order to exit the editing state and complete the editing state
-	const handleBlur = useCallback(() => {
```

Comment:
we were doing a lot of hoops and somersaults to account for what `TldrawUiButtonPicker` should have been managing